### PR TITLE
Core: Improvements around View catalog tests

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 1]
```diff
@@ -132,15 +137,21 @@ public void completeCreateView() {
             .withQuery("trino", "select * from ns.tbl using X")
             .withProperty("prop1", "val1")
             .withProperty("prop2", "val2")
-            .withLocation("file://tmp/ns/view")
+            .withLocation("file:///tmp/ns/view")
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:140]: the file scheme changes are for catalogs that use HadoopFileIO

**Review by ajantha-bhat — APPROVED** [review]
As mentioned in https://github.com/apache/iceberg/pull/7913#discussion_r1363486980 and discussed offline, I too faced these problems when I extend the tests for Nessie Views testing. 

I have now locally verified and confirm that these changes can help for catalogs to extend this class.

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 2]
```diff
@@ -56,10 +59,16 @@ public abstract class ViewCatalogTests<C extends ViewCatalog & SupportsNamespace
 
   protected abstract Catalog tableCatalog();
 
+  @TempDir private Path tempDir;
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:62]: previously we were referring directly to the `tmp` dir (similar to `CatalogTests`), but it's better to use a randomly generated directory. I'll have a follow-up PR to also use a randomly generated temp dir for `CatalogTests`

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 3]
```diff
@@ -664,6 +684,41 @@ public void renameViewTargetAlreadyExistsAsTable() {
         .hasMessageContaining("Cannot rename ns.view to ns.table. Table already exists");
   }
 
+  @Test
+  public void renameTableTargetAlreadyExistsAsView() {
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:688]: this was something that we identified during review of another PR

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 4]
```diff
@@ -225,8 +243,9 @@ public void createViewErrorCases() {
                     .withQuery(trino.dialect(), trino.sql())
                     .withQuery(trino.dialect(), trino.sql())
                     .create())
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessage("Invalid view version: Cannot add multiple queries for dialect trino");
+        .isInstanceOf(Exception.class)
```
Comment (rdblue) [inline core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:246]: I don't think this is correct. We should never allow ANY exception to be valid. Why is `IllegalArgumentException` not the correct one?

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 5]
```diff
@@ -132,15 +144,21 @@ public void completeCreateView() {
...
             .create();
 
     assertThat(view).isNotNull();
     assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+    assertThat(((BaseView) view).operations().current().metadataFileLocation()).isNotNull();
+
+    if (!overridesRequestedLocation()) {
+      assertThat(view.location()).isEqualTo(location);
+    } else {
+      assertThat(view.location()).isNotNull();
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:157]: this goes back to a similar approach that is used for `CatalogTests` (#4906)

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java (hunk 1 above)
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java]: yes this was invalid, but I think it's still better to not directly rely on hardcoded directories under `/tmp` as that might interfere with other tests (running in parallel).

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:157]: also I don't think we fail for tables with a REST catalog when the REST catalog overrides the given location

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java [hunk 6]
```diff
@@ -1446,7 +1544,14 @@ public void updateViewLocationConflict() {
...
-        .isInstanceOf(CommitFailedException.class)
-        .hasMessageContaining("Cannot commit");
+        .satisfiesAnyOf(
+            throwable ->
+                assertThat(throwable)
+                    .isInstanceOf(NoSuchViewException.class)
+                    .hasMessageContaining("View does not exist: ns.view"),
+            throwable ->
+                assertThat(throwable)
+                    .isInstanceOf(CommitFailedException.class)
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:1554]: agreed, I think it's better to standardize on `NoSuchViewException` in this case

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java (hunk 4 above)
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java]: I've removed this change for now. We can discuss it in the REST View support PR

File: core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (rdblue) [reply core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java:156]: Sounds good to me. Thanks for looking into it!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @nastra! Looks like this is also included in #7913 so it's good to get this in.

**Pull request merged**